File: dotnet/niuz.application/entities/Article.cs
namespace niuz.application.entities
{
    public class Article
    {
        public Article(string id, string authorId, string headline)
        {
            Id = id;
            AuthorId = authorId;
            Headline = headline;
        }

        public string Id { get; }
        public string AuthorId { get; }
        public string Headline { get; }

        protected bool Equals(Article other)
        {
            return Id == other.Id;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Article) obj);
        }

        public override int GetHashCode()
        {
            return (Id != null ? Id.GetHashCode() : 0);
        }
    }
}

File: java/src/main/java/com/dddeurope/niuz/newsroom/Article.java
package com.dddeurope.niuz.newsroom;

import java.util.Objects;

public class Article {
    private final String id;
    private final String authorId;
    private final String headline;

    public Article(String id, String authorId, String headline) {
        this.id = id;
        this.authorId = authorId;
        this.headline = headline;
    }

    public String getId() {
        return id;
    }

    public String getAuthorId() {
        return authorId;
    }

    public String getHeadline() {
        return headline;
    }

    @Override
    public String toString() {
        return "Article{" +
                "id='" + id + '\'' +
                ", authorId='" + authorId + '\'' +
                ", headline='" + headline + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Article article = (Article) o;
        return Objects.equals(id, article.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
